File: leetcode/js/the-two-sneaky-numbers-of-digitville.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var getSneakyNumbers = function (nums) {
  const countObj = {};
  const result = [];

  for (const num of nums) {
    countObj[num] = num in countObj ? countObj[num] + 1 : 1;

    if (countObj[num] > 1) {
      result.push(num);

      if (result.length === 2) break;
    }
  }

  return result;
};

File: leetcode/rust/the-two-sneaky-numbers-of-digitville.rs
impl Solution {
    pub fn get_sneaky_numbers(nums: Vec<i32>) -> Vec<i32> {
        use std::collections::HashSet;

        let mut result = Vec::new();
        let mut set = HashSet::new();

        for num in nums {
            match set.contains(&num) {
                true => {
                    result.push(num);
                }
                false => {
                    set.insert(num);
                }
            };
        }

        result
    }
}
